src/components/Map/MapRenderer/MapRenderer.js

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { GoogleMap, Marker, Circle } from '@react-google-maps/api';
+
+const circleOptions = radius => ({
+    strokeColor: 'transparent',
+    strokeOpacity: 1,
+    strokeWeight: 0,
+    fillColor: '#FF0000',
+    fillOpacity: 0.35,
+    clickable: false,
+    draggable: false,
+    editable: false,
+    visible: true,
+    radius,
+    zIndex: 1,
+});
+
+const MapRenderer = React.forwardRef((props, ref) => {
+    const { center, zoom, markers, mapContainerStyle, ...rest } = props;
+    const positions = markers.map(marker => ({ lat: marker.lat, lng: marker.lng }));
```

Comment:
Below in the code there are few places where we iterate `markers` array, but in a loop's body we're getting position of a marker by referencing `positions[i]`. That looks a bit "fragile", cause it assumes that you have to have 2 arrays of the same length where elements relates to each other by index :)

It could be done in a less complicated way just by enriching `markers` array with `position` property in a following way:

```js
const positionedMarkers = markers.map(marker => ({
  ...marker,
  position: {
    lat: marker.lat,
    lng: marker.lng
  }
}));
```

Furthermore, this markers "processing" can be done higher in the `Map` component once. Or, we can even change `markers` prop interface from
```
markers: PropTypes.arrayOf(
    PropTypes.shape({
        lng: PropTypes.number.isRequired,
        lat: PropTypes.number.isRequired,
        radius: PropTypes.number,
    })
),
```
to 

```
markers: PropTypes.arrayOf(
    PropTypes.shape({
        position: PropTypes.shape.isRequired({
            lng: PropTypes.number.isRequired,
            lat: PropTypes.number.isRequired,
        }),
        radius: PropTypes.number,
    })
),
```
and then we don't need to do this processing at all.